Fix 4725 summarize dialog cancel

akankshahu opened the pull request:
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, open one before creating this pull request. -->
Fixes #4725

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. --> 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

akankshahu committed:
Fix #4725: Prevent column addition when summarize dialog is canceled

akankshahu commented:
@Anish9901  Could  you please review it?

pavish committed:
Merge branch 'develop' into fix-4725-summarize-dialog-cancel

pavish reviewed (changes requested):
@akankshahu This PR introduces a regression.

When the user clicks on the "No do not summarize" button, the column should be added without summarization. This is no longer working in this PR.

This is because the `confirm` utility treats both closing the modal and the cancel button in the same way, and this particular use-case repurposes cancel to add the column without summarization.

The right way to do this would be to:
1. Remove the `confirm` utlitiy here
2. Add a separate Modal which looks the same, but both the buttons act as independent actions, one for adding the column without summarization, and the other for adding the column with summarization.
3. Closing that modal should not add the column.

<img width="924" height="581" alt="Image" src="https://github.com/user-attachments/assets/81cd915a-b024-42c3-9a98-0c6fab50d43c" />

akankshahu committed:
Fix #4725: Replace confirm utility with custom modal

- Created SummarizeColumnModal component with independent action buttons
- Removed confirm utility from InputSidebar
- 'Yes' button: adds column with summarization
- 'No' button: adds column without summarization
- Closing modal: does not add column (fixes regression)

akankshahu committed:
Fix code formatting (Prettier)

akankshahu commented:
Hello @pavish ,I tried solving for the issue .Could you please review it?

akankshahu committed:
Fix ESLint errors (import order and trailing spaces)

akankshahu committed:
Fix code formatting in SummarizeColumnModal

pavish committed:
Merge branch 'develop' into fix-4725-summarize-dialog-cancel

pavish reviewed (approved):
@akankshahu The approach and code look good to me! Thanks for your work on this PR.
